libpromises/evalfunction.c

```diff
@@ -5461,6 +5461,8 @@ static FnCallResult FnCallReadFile(ARG_UNUSED EvalContext *ctx, ARG_UNUSED const
     }
     else
     {
+        Log(LOG_LEVEL_ERR, "Function '%s' failed to read file: %s",
+            fp->name, filename);
         return FnFailure();
```

Comment:
I would also skip this since it's changing behaviour, unless somebody requested it. Hmmm after looking at redmine#7702 it seems this even broke expected behaviour in 3.7.2, so we might even revert it there.